### PR TITLE
Configurable UI for Settings to Mods menu and Mods Settings Menu

### PR DESCRIPTION
**Pull request by ModdingPink**

### CODE REVIEW

**Comment by qe201020335** [comment]
Can we not use sirautil by any chance? I don't want to make this simple mod too complex

File: SongRankedBadge/SongRankedBadge.csproj [hunk 1]
```diff
@@ -101,8 +106,15 @@
       <HintPath>$(BeatSaberDir)\Beat Saber_Data\Managed\UnityEngine.VRModule.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Zenject">
+      <HintPath>$(BeatSaberDir)\Beat Saber_Data\Managed\Zenject.dll</HintPath>
+      <Private>False</Private>
+      <SpecificVersion>False</SpecificVersion>
+    </Reference>
+    <Reference Include="Zenject-usage">
+      <HintPath>C:\Program Files (x86)\Steam\steamapps\common\Beat Saber\Beat Saber_Data\Managed\Zenject-usage.dll</HintPath>
```
Comment (qe201020335) [inline SongRankedBadge/SongRankedBadge.csproj]: `BeatSaberDir` Please 😃

Comment (ModdingPink) [reply SongRankedBadge/SongRankedBadge.csproj]: 3:< thats beat saber reference manager being goofy

**Comment by ModdingPink** [comment]
We require SongCore which itself requires SiraUtil, I don't see how it makes a difference

**Comment by qe201020335** [comment]
Oh I did not know that. 
I didn't use SiraUtil originally because the mod was basically just a harmony patch, but now it does make sense to use it for the menu logic. 

Anyways, thanks a lot for the pr!

**Pull request merged**